dockers/docker-database/docker-database-init.sh

```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 
+# Set io-threads to the number of physical CPU cores.
```

Comment:
>physical CPU cores [](http://example.com/codeflow?start=34&length=18)

Seem you are assuming there is only one Redis instance o the host. However we have multi-ASIC use case that there are N Redis instances for N ASICs.

Then you set io-threads to N*cores. Will it harm the performance?

Reply:
Reading the conf file document

```
# By default threading is disabled, we suggest enabling it only in machines
# that have at least 4 or more cores, leaving at least one spare core.
# Using more than 8 threads is unlikely to help much. We also recommend using
# threaded I/O only if you actually have performance problems, with Redis
# instances being able to use a quite big percentage of CPU time, otherwise
# there is no point in using this feature.
#
# So for instance if you have a four cores boxes, try to use 2 or 3 I/O
# threads, if you have a 8 cores, try to use 6 threads. In order to
# enable I/O threads use the following configuration directive:
```
Even for single Redis instance, it is recommended to leave spare core.

Reply:
After some testing, we've now decided to disable the multi-threading in multi-ASIC platforms and limit the thread numbers to between 2 to 8 threads.

Reply:
the threads logic is like below:

```python
# Retrieve total number of ASICs in the platform
num_asic = get_asic_nums()

# Retrieve total number of CPU cores in the platform
num_core = get_total_cores()

if num_asic > 1:
    # the multi-threading will be disabled in multi ASIC platform
    threads = 1
elif num_core < 4:
    # the multi-threading will be disabled if less than 4 cores
    threads = 1
else:
    threads = min(num_core//2, 8)

if threads > 1:
    enable_io_threads()
```